app/assets/javascripts/discourse/app/controllers/user-notifications.js

```diff
@@ -2,22 +2,51 @@ import Controller from "@ember/controller";
 import { inject as service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import DismissNotificationConfirmationModal from "discourse/components/modal/dismiss-notification-confirmation";
+import RelativeDate from "discourse/components/relative-date";
 import { ajax } from "discourse/lib/ajax";
+import UserMenuNotificationItem from "discourse/lib/user-menu/notification-item";
 import getURL from "discourse-common/lib/get-url";
 import { iconHTML } from "discourse-common/lib/icon-library";
 import discourseComputed from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
 
 export default Controller.extend({
   modal: service(),
+  appEvents: service(),
+  currentUser: service(),
+  siteSettings: service(),
+  site: service(),
+
   queryParams: ["filter"],
   filter: "all",
 
+  get listContainerClassNames() {
+    return `user-notifications-list ${
+      this.siteSettings.show_user_menu_avatars ? "show-avatars" : ""
+    }`;
+  },
+
   @discourseComputed("filter")
   isFiltered() {
     return this.filter && this.filter !== "all";
   },
 
+  @discourseComputed("model.content.@each")
+  items() {
+    return this.model.map((notification) => {
+      const props = {
+        appEvents: this.appEvents,
+        currentUser: this.currentUser,
+        siteSettings: this.siteSettings,
+        site: this.site,
+        notification,
+        endComponent: RelativeDate,
```

Comment:
If we rename this file to `.gjs`, then we can do this:
```suggestion
        endComponent: <template><RelativeDate @date={{@data.date}} /></template>
```
and thereby remove the need for the weird `this.args.data.date` in the `RelativeDate` component.

Reply:
WAAAAAt

Reply:
Or maybe we can go even fancier and read the notification straight from the local scope, and then remove the need for `endComponentData` ? 🤔 

```
endComponent: <template><RelativeDate @date={{notification.created_at}} /></template>
```

Reply:
Yeah exactly hehe wooo you bad boy!